core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -81,155 +95,171 @@ public void testAddDeleteFile() {
-                .validateDataFilesExist(ImmutableList.of(FILE_A.path()))
-                .commit());
+            commit(
+                table,
+                table
+                    .newRowDelta()
+                    .addDeletes(FILE_A_DELETES)
+                    .validateFromSnapshot(validateFromSnapshotId)
+                    .validateDataFilesExist(ImmutableList.of(FILE_A.path())),
+                branch));
 
     Assert.assertEquals(
         "Table state should not be modified by failed RowDelta operation",
         deleteSnapshotId,
-        table.currentSnapshot().snapshotId());
+        latestSnapshot(table, branch).snapshotId());
 
     Assert.assertEquals(
         "Table should not have any delete manifests",
         0,
-        table.currentSnapshot().deleteManifests(table.io()).size());
+        latestSnapshot(table, branch).deleteManifests(table.io()).size());
   }
 
   @Test
   public void testValidateDataFilesExistReplacePartitions() {
-    table.newAppend().appendFile(FILE_A).appendFile(FILE_B).commit();
+    commit(table, table.newAppend().appendFile(FILE_A).appendFile(FILE_B), branch);
 
     // test changes to the table back to the snapshot where FILE_A and FILE_B existed
-    long validateFromSnapshotId = table.currentSnapshot().snapshotId();
+    long validateFromSnapshotId = latestSnapshot(table, branch).snapshotId();
 
     // overwrite FILE_A's partition
-    table.newReplacePartitions().addFile(FILE_A2).commit();
+    commit(table, table.newReplacePartitions().addFile(FILE_A2), branch);
 
-    long deleteSnapshotId = table.currentSnapshot().snapshotId();
+    long deleteSnapshotId = latestSnapshot(table, branch).snapshotId();
 
     AssertHelpers.assertThrows(
         "Should fail to add FILE_A_DELETES because FILE_A is missing",
         ValidationException.class,
         "Cannot commit, missing data files",
         () ->
-            table
-                .newRowDelta()
-                .addDeletes(FILE_A_DELETES)
-                .validateFromSnapshot(validateFromSnapshotId)
-                .validateDataFilesExist(ImmutableList.of(FILE_A.path()))
-                .commit());
+            commit(
+                table,
+                table
+                    .newRowDelta()
+                    .addDeletes(FILE_A_DELETES)
+                    .validateFromSnapshot(validateFromSnapshotId)
+                    .validateDataFilesExist(ImmutableList.of(FILE_A.path())),
+                branch));
 
     Assert.assertEquals(
         "Table state should not be modified by failed RowDelta operation",
         deleteSnapshotId,
-        table.currentSnapshot().snapshotId());
+        latestSnapshot(table, branch).snapshotId());
 
     Assert.assertEquals(
         "Table should not have any delete manifests",
         0,
-        table.currentSnapshot().deleteManifests(table.io()).size());
+        latestSnapshot(table, branch).deleteManifests(table.io()).size());
   }
 
-  @Test
+  // @Test
```

Comment:
@amogh-jahagirdar, why is this test disabled?

Reply:
This passes locally. I'm going to assume this was an accident and restore the test to get CI running it.

Reply:
yeah shouldn't be disabled, thanks!